event_routing_backends/processors/xapi/openedx_filters/filters.py

```diff
@@ -0,0 +1,40 @@
+"""
+XAPI filters, this file aims to contain all the filters that could modify the
+standard transformer results by implementing external pipeline steps.
+"""
+from openedx_filters.tooling import OpenEdxPublicFilter
+
+from event_routing_backends.processors.xapi.openedx_filters.exceptions import XApiInvalidFilterType
+
+
+class XApiBaseFilter(OpenEdxPublicFilter):
+    """
+    This is a general filter class that applies the open edx filter in multiple
+    scenarios, its functionality is limited to one input one output therefore this
+    only can be applied in method that returns a unique value.
+    """
+
+    @classmethod
+    def set_filter_type(cls, filter_type):
+        """ Set the class attribute `filter_type`
+
+        Arguments
+            filter_type: String the defines the filter key on the OPEN_EDX_FILTERS_CONFIG
+                section
+        """
+        cls.filter_type = filter_type
```

Comment:
Why here is necessary to add the `filter_type` to the class if the class is not initialized(assigned). So this gonna change every time for the class.
 Also this  is necessary to be defined before using the class method `run_filter`. That is the intention or doesnt make sense that run_filter receive and extra arg like `filter_type`???

Reply:
That attribute is used by the openedx-filters code and allows to identify the filter  in the settings, general implementation of filter is to create a class with a fixed value this implementation is dynamic therefore it's not necessary to create a class per filter, this allows to reuse the same class by changing that